### ISSUE TITLE
No notification / Polling messages

### ISSUE THREAD

**Issue by ramack** [mattermost/mattermost-mobile#3355]
#### Summary
When a new message come in, I don't get any notification from the mattermost beta app.

#### Environment Information

- Device Name: Xcover 4
- OS Version: Android Pie
- Mattermost App Version: 1.19.0 (Build 193)
- Mattermost Server Version: 5.2.0 (Build 5.2.1)

#### Steps to reproduce
Disable all google services.

#### Expected behavior
Notification on new messages.

#### Observed behavior
No notifications shown by the app. When manually opening new messages in public and 1:1 communication channels are shown.

#### Possible fixes
As no GCM service is available the app should switch to a polling mode, where with a forground service the app polls in configurable time whether there are new messages.

**Comment by amyblais** [comment]
@ramack Would you be open to updating to latest versions (mobile 1.23.1 and server 5.15) to see if you experience the same issue?

**Comment by enahum** [comment]
@ramack no polling mechanism is present and currently we are not planning to add it.

If this is a major pain point for you and you team you can consider joining our [community server](https://community.mattermost.com] and have a conversation about this to figure out the proper implementation that then you and your team could work on and contribute to the project if determined that it makes sense.

**Comment by ramack** [comment]
> @ramack Would you be open to updating to latest versions (mobile 1.23.1 and server 5.15) to see if you experience the same issue?

For sure, I'll update automatically as soon as the version is available in f-droid. Did you setup the detection of new versions correctly?

**Comment by enahum** [comment]
Latest version does not have a polling mechanism and sadly no other version will as long as it is not contributed by someone.

Also updates to f-droid are done by the community and not the mattermost team.

**Comment by ramack** [comment]
> @ramack no polling mechanism is present and currently we are not planning to add it.
> 
> If this is a major pain point for you and you team you can consider joining our [community server]([https://community.mattermost.com]](https://community.mattermost.com%5D) and have a conversation about this to figure out the proper implementation that then you and your team could work on and contribute to the project if determined that it makes sense.

If you don't want to support devices that are free of google services want is the meaning of doing the app as free software? I can live with manual polling it will just not be as instant as what mattermost promises. My tesm is aware that mattermost is not relyable in notifications.

I mainly filed this ticket for you as an opportunity to improve, if you don't care, I don't also.

**Comment by enahum** [comment]
We never said we don’t care.. we are open to discuss it and a Pull request is more than welcome ;)

**Comment by ramack** [comment]
> Also updates to f-droid are done by the community and not the mattermost team.

Yes, I have seen. But it doesn't look easy to build your app, so maybe you want to help a little out, because you as developers have two options.
1) Do your best to make it as easy as possible to build for f-droid or
2) leave everything up to some poor guys trying to make your app available for people who care about really free software and/or have some concerns with google.

> We never said we don’t care.. we are open to discuss it and a Pull request is more than welcome ;)

You didn't say that explicitly, but when I read your answer it felt a little like that. Clearly we all want contributions from users to our projects and we have to prioritize between different requests, but the way you say it has a big impact on the decision whether someone stays and actively contributes or just goes away. - There are so many opportunities to contribute out there, any - I guess - nobody is waiting to be able to contribute to exactly your project. And if someone takes the time to create an issue on github with a feature request he is already somehow interested...

**Comment by 9cat** [comment]
please nehance the notication , it is really the worst thing in mattermost. we really want to heavily depend on mattermost. but he notification part is really not that satisified for a long time.

**Comment by enahum** [comment]
@9cat Sorry for the delayed response, but can you please share with us what is it that you are seeing and what is your expectation?
FYI @esethna

**Comment by jasonblais** [comment]
@9cat sorry to hear about your experience. Would you be open to share more on the behaviour you're observing, and what you would expect to see?

**Comment by Mannshoch** [comment]
Unified Push support: https://github.com/mattermost/mattermost-mobile/issues/6673
